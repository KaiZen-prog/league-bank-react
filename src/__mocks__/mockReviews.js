export const mockReviews = [{
  id: 1,
  author: {
    name: 'Cтепан',
    surname: 'Иванов'
  },
  rating: 4,
  text: 'Просто восхитительный банк!',
  date: '2022-11-15T13:58:19.494Z',
},

  {
    id: 2,
    author: {
      name: 'Иван',
      surname: 'Cтепанов'
    },
    rating: 3,
    text: 'Чат-бот техподдержки плохо понимает запросы',
    date: '2022-10-15T13:58:19.494Z',
  }
];
