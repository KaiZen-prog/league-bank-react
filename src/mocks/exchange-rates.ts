export const exchangeRates = {
  '2024-01-19': {
    USD: 1,
    RUB: 88.573959,
    EUR: 0.918438,
    KZT: 452.633544,
    GBP: 0.787512,
    CNY: 7.1197
  },
  '2024-01-18': {
    USD: 1,
    RUB: 88.638497,
    EUR: 0.918649,
    KZT: 453.345273,
    GBP: 0.788835,
    CNY: 7.1263
  },
  '2024-01-17': {
    USD: 1,
    RUB: 88.105727,
    EUR: 0.919239,
    KZT: 453.139857,
    GBP: 0.791321,
    CNY: 7.1265
  },
  '2024-01-16': {
    USD: 1,
    RUB: 87.719298,
    EUR: 0.913662,
    KZT: 451.339299,
    GBP: 0.786436,
    CNY: 7.1073
  },
  '2024-01-15': {
    USD: 1,
    RUB: 88.043653,
    EUR: 0.913639,
    KZT: 451.160232,
    GBP: 0.785191,
    CNY: 7.111
  },
  '2024-01-14': {
    USD: 1,
    RUB: 87.959997,
    EUR: 0.912076,
    KZT: 451.160232,
    GBP: 0.78456,
    CNY: 7.1126
  },
  '2024-01-13': {
    USD: 1,
    RUB: 87.959997,
    EUR: 0.912076,
    KZT: 451.530905,
    GBP: 0.78456,
    CNY: 7.1106
  },
  '2024-01-12': {
    USD: 1,
    RUB: 88.495575,
    EUR: 0.910544,
    KZT: 451.995747,
    GBP: 0.782515,
    CNY: 7.09752
  }
};
