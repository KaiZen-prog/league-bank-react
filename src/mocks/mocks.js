export const promoSlides = [
  `img/desktop-slide-1.jpg`,
  `img/desktop-slide-2.jpg`,
  `img/desktop-slide-3.jpg`,
];
