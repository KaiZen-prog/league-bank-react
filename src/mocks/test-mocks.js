import moment from "moment";

export const InitialState = {
  CONVERSION_HISTORY: {
    conversionHistory: [
      {
        date: `18.09.2021`,
        currencyInput: {
          type: `USD`,
          amount: 6549
        },
        currencyOutput: {
          type: `EUR`,
          amount: 5585.07
        }
      },
      {
        date: `19.09.2021`,
        currencyInput: {
          type: `CNY`,
          amount: 23
        },
        currencyOutput: {
          type: `RUB`,
          amount: 258.59
        }
      },
      {
        date: `20.09.2021`,
        currencyInput: {
          type: `EUR`,
          amount: 12458
        },
        currencyOutput: {
          type: `GBP`,
          amount: 10694.16
        }
      },
      {
        date: `13.09.2021`,
        currencyInput: {
          type: `CNY`,
          amount: 654
        },
        currencyOutput: {
          type: `RUB`,
          amount: 7363.93
        }
      },
      {
        date: `15.09.2021`,
        currencyInput: {
          type: `EUR`,
          amount: 105
        },
        currencyOutput: {
          type: `GBP`,
          amount: 89.6
        }
      },
      {
        date: `20.09.2021`,
        currencyInput: {
          type: `RUB`,
          amount: 105
        },
        currencyOutput: {
          type: `USD`,
          amount: 1.43
        }
      }
    ]
  },

  CONVERTER: {
    date: moment().utc().format(`YYYY-MM-DD`),
    exchangeRate: {
      USD: 1,
      RUB: 72.8797,
      EUR: 0.852813,
      GBP: 0.727776,
      CNY: 6.4662
    }
  }
};
