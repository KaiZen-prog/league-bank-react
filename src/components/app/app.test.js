import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import {InitialState} from '../../__mocks__/mocks';
import configureStore from 'redux-mock-store';

const mockStore = configureStore();
import App from "./app";

test(`App render correctly`, () => {
  const store = mockStore(InitialState);

  const tree = renderer
      .create(
          <Provider store={store}>
            <App/>
          </Provider>
      )
      .toJSON();
  expect(tree).toMatchSnapshot();
});
