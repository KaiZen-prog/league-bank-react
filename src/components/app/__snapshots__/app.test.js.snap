// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render correctly 1`] = `
Array [
  <header
    className="header  app__header"
  >
    <div
      className="container header__container"
    >
      <div
        className="header__wrapper"
      >
        <button
          className="header__burger-button"
          onClick={[Function]}
        >
          <span
            className="visually-hidden"
          >
            Открыть меню
          </span>
        </button>
        <a
          className="header__logo"
          href="/"
          onClick={[Function]}
        >
          ЛИГА Банк
        </a>
        <button
          className="header__close-menu "
          onClick={[Function]}
        >
          <span
            className="visually-hidden"
          >
            Закрыть меню
          </span>
        </button>
      </div>
      <nav
        className="header__nav "
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item"
          >
            <a
              className="header__nav-link "
              href="#"
            >
              Услуги
            </a>
          </li>
          <li
            className="header__nav-item"
          >
            <a
              className="header__nav-link "
              href="#"
            >
              Рассчитать кредит
            </a>
          </li>
          <li
            className="header__nav-item"
          >
            <a
              className="header__nav-link "
              href="/converter"
              onClick={[Function]}
            >
              Конвертер валют
            </a>
          </li>
          <li
            className="header__nav-item"
          >
            <a
              className="header__nav-link "
              href="#"
            >
              Контакты
            </a>
          </li>
        </ul>
      </nav>
      <div
        className="header__user-block "
      >
        <a
          className="header__user-link "
          href="#"
          onClick={[Function]}
        >
          <span
            className="header__user-link-value "
          >
            Войти в Интернет-банк
          </span>
        </a>
        <div
          className="log-in "
          onClick={[Function]}
        >
          <form
            action="#"
            className="log-in__form"
            onClick={[Function]}
          >
            <div
              className="log-in__logo"
            />
            <button
              className="log-in__close"
              onClick={[Function]}
              type="button"
            >
              <span
                className="visually-hidden"
              >
                Закрыть окно
              </span>
            </button>
            <label
              className="log-in__label log-in__label--login"
            >
              Логин
              <input
                autoFocus={true}
                className="log-in__input log-in__input--login"
                id="login"
                name="login"
                onChange={[Function]}
                required={true}
                type="text"
                value=""
              />
            </label>
            <label
              className="log-in__label log-in__label--password"
            >
              Пароль
              <input
                className="log-in__input log-in__input--password"
                id="password"
                name="password"
                onChange={[Function]}
                required={true}
                type="password"
                value=""
              />
            </label>
            <button
              className="log-in__show-password"
              onMouseDown={[Function]}
              onMouseUp={[Function]}
            />
            <button
              className="log-in__submit"
              type="submit"
            >
              Войти
            </button>
            <a
              className="log-in__restore-password"
              href="#top"
            >
              Забыли пароль?
            </a>
          </form>
        </div>
      </div>
    </div>
  </header>,
  <main>
    <section
      className="main-slider"
    >
      <div
        className="main-slider__slides-container"
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "left": "0",
          }
        }
      />
      <ul
        className="main-slider__dots"
      />
    </section>
    <section
      className="services-slider"
    >
      <h2
        className="services-slider__header visually-hidden"
      >
        Наши сервисы
      </h2>
      <ul
        className="services-slider__tabs"
      >
        <li
          className="services-slider__tab services-slider__tab--deposit "
          onClick={[Function]}
        >
          <span
            className="services-slider__tab-label services-slider__tab-label--deposit"
          >
            Вклады
          </span>
        </li>
        <li
          className="services-slider__tab services-slider__tab--credit "
          onClick={[Function]}
        >
          <span
            className="services-slider__tab-label services-slider__tab-label--credit"
          >
            Кредиты
          </span>
        </li>
        <li
          className="services-slider__tab services-slider__tab--insurance "
          onClick={[Function]}
        >
          <span
            className="services-slider__tab-label services-slider__tab-label--insurance"
          >
            Страхование
          </span>
        </li>
        <li
          className="services-slider__tab services-slider__tab--online "
          onClick={[Function]}
        >
          <span
            className="services-slider__tab-label services-slider__tab-label--online"
          >
            Онлайн-сервисы
          </span>
        </li>
      </ul>
      <div
        className="services-slider__slides-container"
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "left": "0",
          }
        }
      />
      <ul
        className="services-slider__dots"
      />
    </section>
    <section
      className="calculator"
    >
      <a
        name="calculator"
      />
      <form
        action="#"
        className="calculator__form"
        onSubmit={[Function]}
      >
        <h2
          className="calculator__title"
        >
          Кредитный калькулятор
        </h2>
        <div
          className="calculator__flex-container"
        >
          <div
            className="calculator__container"
          >
            <fieldset
              className="calculator__purpose"
            >
              <h3
                className="calculator__step"
              >
                Шаг 1. Цель кредита
              </h3>
              <div
                className="calculator__purpose-select"
                onClick={[Function]}
              >
                <span
                  className="calculator__select-title"
                >
                  Выберите цель кредита
                </span>
                <ul
                  className="calculator__purpose-list calculator__purpose-list--hidden"
                >
                  <li
                    className="calculator__purpose-item"
                    id="mortgage"
                    onClick={[Function]}
                  >
                    Ипотечное кредитование
                  </li>
                  <li
                    className="calculator__purpose-item"
                    id="car"
                    onClick={[Function]}
                  >
                    Автомобильное кредитование
                  </li>
                </ul>
              </div>
            </fieldset>
          </div>
        </div>
      </form>
    </section>
    <section
      className="map"
      id="map"
    >
      <h2
        className="map__header"
      >
        Отделения Лига Банка
      </h2>
      <iframe
        className="map__iframe"
        src="https://yandex.ru/map-widget/v1/?um=constructor%3A903677e276cd647a44c0897e60edad9ae34a0ebe73fde4d37f4ca90b4001c09f&source=constructor"
        title="map"
      />
    </section>
  </main>,
  <footer
    className="footer app__footer"
  >
    <div
      className="footer__container"
    >
      <section
        className="footer__address"
      >
        <a
          className="footer__logo"
          href="#"
        >
          ЛИГА Банк
        </a>
        <div
          className="footer__info"
        >
          <p>
            150015, г. Москва, ул. Московская, д. 32
          </p>
          <p>
            Генеральная лицензия Банка России №1050
          </p>
          <p>
            Ⓒ Лига Банк, 2019
          </p>
        </div>
        <nav
          className="footer__nav footer-nav"
        >
          <ul
            className="footer-nav__list"
          >
            <li
              className="footer-nav__item"
            >
              <a
                className="footer-nav__link"
                href="#"
              >
                Услуги
              </a>
            </li>
            <li
              className="footer-nav__item"
            >
              <a
                className="footer-nav__link"
                href="#"
              >
                Рассчитать кредит
              </a>
            </li>
            <li
              className="footer-nav__item"
            >
              <a
                className="footer-nav__link"
                href="#"
              >
                Контакты
              </a>
            </li>
            <li
              className="footer-nav__item"
            >
              <a
                className="footer-nav__link"
                href="#"
              >
                Задать вопрос
              </a>
            </li>
          </ul>
        </nav>
      </section>
      <section
        className="footer__contacts"
      >
        <ul
          className="footer__contacts-list"
        >
          <li
            className="footer__contacts-item"
          >
            <a
              className="footer__tel footer__tel--mobile"
              href="tel: *0904"
            >
              *0904
            </a>
            <p
              className="footer__phone-info"
            >
              Бесплатно для абонентов МТС, Билайн, Мегафон, Теле2
            </p>
          </li>
          <li
            className="footer__contacts-item footer__contacts-item--main-phone"
          >
            <a
              className="footer__tel footer__tel--main"
              href="tel: +78001112233"
            >
              8 800 111 22 33
            </a>
            <p
              className="footer__phone-info"
            >
              Бесплатный для всех городов России
            </p>
          </li>
        </ul>
        <section
          className="footer__social social"
        >
          <ul
            className="social__list"
          >
            <li
              className="social__item"
            >
              <a
                className="social__link social__link--fb"
                href="#"
              >
                <span
                  className="visually-hidden"
                >
                  Мы в фейсбуке
                </span>
              </a>
            </li>
            <li
              className="social__item"
            >
              <a
                className="social__link social__link--inst"
                href="#"
              >
                <span
                  className="visually-hidden"
                >
                  Мы в инстаграме
                </span>
              </a>
            </li>
            <li
              className="social__item"
            >
              <a
                className="social__link social__link--twitter"
                href="#"
              >
                <span
                  className="visually-hidden"
                >
                  Мы в твиттере
                </span>
              </a>
            </li>
            <li
              className="social__item"
            >
              <a
                className="social__link social__link--youtube"
                href="#"
              >
                <span
                  className="visually-hidden"
                >
                  Наш канал на ютубе
                </span>
              </a>
            </li>
          </ul>
        </section>
      </section>
    </div>
  </footer>,
]
`;
